### DISCUSSION TITLE
Timeout Error

### DISCUSSION THREAD

**Discussion by CalFamily** [winder/Universal-G-Code-Sender#2592] [Help]
I'm new to CNC so be gentle!!

I've got my machine connected and can run a project. There have been a few times I've needed to stop the project before the end for one reason or another. When I hit the STOP button, the progress pauses. When I hit it again to cancel, I get the message "[Error] Timeout waiting for GRBL to Idle during cancel, cancel incomplete." In order to abort a project I have to pause, then disconnect the machine and reconnect. That doesn't work well when the reason for a pause is a bit change within the project because when I reconnect the work coordinate zero isn't the same.

Any suggestions?

**Comment by breiler** [comment]
What type of controller are you using? What firmware are you using (for instance GRBL 1.1f)? What version of UGS have you installed?

**Comment by CalFamily** [comment]
I just use UGS onboard controls no external or add on controllers. The
firmware I have selected is GRBL and it’s the only one in the drop down
menu. If there is anything else that I should be using I don’t k ow about
it (like I said, I’m super new to this). I think I have the most up to date
version. When I check for updates it says I’m up to date. Under the about
it says “Product Version UGS Platform 20300207”

**Reply by breiler** [reply]
> I just use UGS onboard controls no external or add on controllers. The
firmware I have selected is GRBL and it’s the only one in the drop down
menu.

I want to know the version of GRBL you have installed

> Under the about it says “Product Version UGS Platform 20300207”

The numbers should form a date as YYYYMMDD, can you take a screenshot and upload it here.

**Comment by CalFamily** [comment]
I’m not sure if this is what you’re looking for. When I connect to the
machine this is what comes up. If this isn’t what you need where do I find
the version of GRBL?

**Reply by breiler** [reply]
If you tried to attach a image via email that doesn't work. You need to attach it on GitHub.

**Reply by CalFamily** [reply]
![IMG_1129](https://github.com/user-attachments/assets/1376382e-5317-4e3f-87a6-55ac8b555f2a)

hopefully that works

**Reply by breiler** [reply]
Ok that is the later version of GRBL so that should be ok.

What version of UGS, take a screen shot of the entire application.

**Comment by CalFamily** [comment]
<img width="926" alt="image" src="https://github.com/user-attachments/assets/12780f60-b1d8-49d4-ad7f-bb94b3ff29ff">

**Reply by breiler** [reply]
That is the UGS version supplied from MillRight, and it looks ancient.

You should try to upgrade to the latest version here: https://universalgcodesender.com/download/

**Comment by CalFamily** [comment]
Hello!

My mother board on my computer had an unfortunate accident and it’s taken a
minute to get a new computer. I’ve downloaded the new version of UGS. Now
the problem I’m running into is that when I import my gcode file from
carbide create, nothing shows up in UGS. It has no toolpaths for the design
I created so now I can’t cut anything at all. Any suggestions for that?
